### PR TITLE
Increase max dash length limit and Make cable cars and power lines dash patterns longer 

### PR DESCRIPTION
**Pull request by pastk**
Fixes https://github.com/organicmaps/organicmaps/pull/4711#issuecomment-1458319703

The limit was 32 (for an individual dash), now its 128 (cumulative for both dashes) and limited by the texture size.

### CODE REVIEW

**Review by vng — APPROVED** [review]

**Comment by pastk** [comment]
Rebased and removed one added assert, because it was duplicate, actually. 
And the limit is 127 actually because 2 pixels of the texture are reserved for some reasom.

File: drape/stipple_pen_resource.hpp [hunk 1]
```diff
@@ -17,12 +17,14 @@
...
 {
-// Based on ./data/patterns.txt, all paterns have 2 entries now.
-using PenPatternT = buffer_vector<uint8_t, 2>;
+uint32_t constexpr kMaxStipplePenLength = 512;  /// @todo Should be equal with kStippleTextureWidth?
 
-inline uint8_t PatternFloat2Pixel(double d)
+// Based on ./data/patterns.txt, the most of patterns have 2 entries (4 entries for triangles pattern).
+using PenPatternT = buffer_vector<uint16_t, 2>;
+
+inline uint16_t PatternFloat2Pixel(double d)
```
Comment (biodranik) [inline drape/stipple_pen_resource.hpp:25]:
```suggestion
inline constexpr uint16_t PatternFloat2Pixel(double d)
```

File: drape_frontend/line_shape.cpp [hunk 2]
```diff
@@ -364,7 +364,7 @@ void LineShape::Construct<DashedLineBuilder>(DashedLineBuilder & builder) const
   // Each segment should lie in pattern mask according to the "longest" possible pixel length in current tile.
   // Since, we calculate vertices once, usually for the "smallest" tile scale, need to apply divide factor here.
   // In other words, if m_baseGtoPScale = Scale(tileLevel), we should use Scale(tileLevel + 1) to calculate 'maskLengthG'.
-  /// @todo Logically, the factor should be 2, but drawing artifacts still present.
+  /// @todo Logically, the factor should be 2, but drawing artifacts still present at higher visual scales.
```
Comment (biodranik) [inline drape_frontend/line_shape.cpp:367]:
```suggestion
  /// @todo Logically, the factor should be 2, but drawing artifacts are still present at higher visual scales.
```

**Review by biodranik — COMMENTED** [review]
1. Where is the 127 limit documented now?
2. How the changed types are rendered now? Do they look good?

File: drape/stipple_pen_resource.hpp (hunk 1 above)
Comment (pastk) [reply drape/stipple_pen_resource.hpp:25]: not sure it'll work with NDK, but let's try...

**Comment by pastk** [comment]
> 1. Where is the 127 limit documented now?

ATM nowhere, just like the old limit was not documented.

> 
>     2. How the changed types are rendered now? Do they look good?

Yes. Basically its the first batch of screenshots in #4711 (made powerlines pattern a tad more longer).

**Review by biodranik — APPROVED** [review]

File: drape/stipple_pen_resource.hpp (hunk 1 above)
Comment (pastk) [reply drape/stipple_pen_resource.hpp]: works fine!

**Comment by pastk** [comment]
Added @biodranik's suggestions and rebased.

**Review by biodranik — APPROVED** [review]

**Pull request merged**